### PR TITLE
Upgrade Kentik API client and minor improvements

### PR DESCRIPTION
**Pull request by Fenthick**
Additional improvements:
- Mark resource in TF state as deleted if read request returns HTTP 404.
  This change ensures that TF state is consistent with the actual state.
- Remove read request check in delete operation because it is no longer
  needed
- Run top-level tests in parallel. It reduced test time from ~40 s to
  ~12 s on my machine
- Add log_payloads config option
- Refactor provider.go so that it is similar to the one in Synthetics
  provider and easier to maintain

Issue: https://clkentik.atlassian.net/browse/KNTK-332

### CODE REVIEW

**Comment by Opelord** [comment]
Looks good to me, all my questions are answered in your summary.

**Review by Opelord — DISMISSED** [review]

File: internal/provider/provider.go [hunk 1]
```diff
@@ -2,70 +2,88 @@ package provider
...
 				DefaultFunc: schema.EnvDefaultFunc("KTAPI_AUTH_EMAIL", nil),
-				Description: "Authorization. Either email attribute or KTAPI_AUTH_EMAIL environment variable is required",
+				Description: "Authorization email (required). Can also be specified with KTAPI_AUTH_EMAIL environment variable.",
 			},
-			"token": &schema.Schema{
+			tokenKey: {
 				Type:        schema.TypeString,
 				Sensitive:   true,
-				Required:    true,
+				Optional:    true,
```
Comment (mmac-m3a) [inline internal/provider/provider.go:42]: I'm not entirely sure what the `Optional` and `Required` attributes mean here. However if the `emailKey` has `Required` set to true, so should the `tokenKey`. Both email and token must be provided (one way or another) in order for the provider to be able to access the API.

**Review by mmac-m3a — CHANGES_REQUESTED** [review]
AFAIK, both `email` and `token` are required. If not, please, explain why/how?

File: internal/provider/provider.go (hunk 1 above)
Comment (Fenthick) [reply internal/provider/provider.go]: Thank you for catching that. I was experimenting with required attributes when I was working on plan_id task in AWS automation and I reverted the changes wrong.

**Review by mmac-m3a — APPROVED** [review]
LGTM

**Pull request merged**